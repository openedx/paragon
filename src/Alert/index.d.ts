import React from 'react';
import BaseAlert, { AlertProps } from 'react-bootstrap/Alert';

declare const ALERT_CLOSE_LABEL_TEXT: string;
export interface BaseAlertProps extends AlertProps {
  icon?: React.ReactElement,
  actions?: Array<React.ReactElement>,
  dismissible?: boolean,
  onClose?: () => void,
  closeLabel?: React.ReactElement,
  stacked?: boolean,
}
declare const Alert: React.ForwardRefExoticComponent<BaseAlertProps> & {
  Link: typeof BaseAlert.Link;
  Heading: typeof BaseAlert.Heading;
};
export { ALERT_CLOSE_LABEL_TEXT };
export default Alert;
