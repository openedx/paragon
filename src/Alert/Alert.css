/**
 * Do not edit directly
 * Generated on Wed, 01 Mar 2023 00:27:19 GMT
 */

:root {
  --pgn-typography-alert-line-height: 1.5rem;
  --pgn-typography-alert-font-size: .875rem;
  --pgn-spacing-alert-icon-space: .8rem;
  --pgn-spacing-alert-margin-bottom: 1rem;
  --pgn-spacing-alert-padding-x: 1.5rem;
  --pgn-spacing-alert-padding-y: 1.5rem;
  --pgn-size-alert-border-width: 0;
  --pgn-typography-alert-font-weight-link: var(--pgn-typography-font-weight-normal);
  --pgn-size-alert-border-radius: var(--pgn-size-border-radius-base);
  --pgn-spacing-alert-actions-gap: var(--pgn-spacing-spacer-3);
}
