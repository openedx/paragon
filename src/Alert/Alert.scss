//
// Override Bootstrap's alert definition.
// We do this to get more control over colors
// used in alerts.
//

//
// Base styles
//

.alert {
  position: relative;
  padding: var(--pgn-spacing-alert-padding-y) var(--pgn-spacing-alert-padding-x);
  margin-bottom: var(--pgn-spacing-alert-margin-bottom);
  border: var(--pgn-size-alert-border-width) solid var(--pgn-alert-border-color, transparent);
  color: inherit;
  background-color: var(--pgn-alert-bg, transparent);

  @include border-radius(var(--pgn-size-alert-border-radius));
  @include pgn-box-shadow(1, "down");

  .alert-message-content > :last-child {
    margin-bottom: 0;
  }

  .alert-icon {
    color: var(--pgn-alert-icon-color, inherit);
    float: left;
    margin-inline-end: var(--pgn-spacing-alert-icon-space);
    width: 1.5rem;
    height: 1.5rem;
    flex-shrink: 0;
  }
}

// Headings for larger alerts
.alert-heading {
  // Specified to prevent conflicts of changing var(--pgn-color-headings-base)
  color: var(--pgn-color-alert-title);
  display: flex;
}

// added to align content and icon
.alert-content {
  display: flex;
  font-size: var(--pgn-typography-alert-font-size);
  line-height: var(--pgn-typography-alert-line-height);
  color: var(--pgn-color-alert-content);
  text-align: start;

  div {
    flex-grow: 1;
  }
}

// Provide class for links that match alerts
.alert-link {
  font-weight: var(--pgn-typography-alert-font-weight-link);
  text-decoration: underline;

  &:hover {
    text-decoration: none;
  }
}

// Dismissible alerts
//
// Expand the right padding and account for the close button's positioning.

// Baking in $close-font-size: var(--pgn-typography-font-size-base) * 1.5 !default; to avoid any dependency
.alert-dismissible {
  padding-right: calc((var(--pgn-typography-font-size-base) * 1.5) + (var(--pgn-spacing-alert-padding-x) * 2));

  // Adjust close link position
  .close {
    position: absolute;
    top: 0;
    right: 0;
    padding: var(--pgn-spacing-alert-padding-y) var(--pgn-spacing-alert-padding-x);
    color: inherit;
  }
}

.pgn__alert-message-wrapper {
  display: flex;
  align-items: center;

  .pgn__alert-actions {
    margin-inline-start: var(--pgn-spacing-alert-actions-gap);
  }
}

.pgn__alert-message-wrapper-stacked .pgn__alert-actions {
  margin-top: var(--pgn-spacing-alert-actions-gap);
}

// Alternate styles

.alert-success {
  --pgn-alert-bg: var(--pgn-color-alert-bg-success);
  --pgn-alert-border-color: var(--pgn-color-alert-border-success);
  --pgn-alert-icon-color: var(--pgn-color-alert-icon-success);
}

.alert-info {
  --pgn-alert-bg: var(--pgn-color-alert-bg-info);
  --pgn-alert-border-color: var(--pgn-color-alert-border-info);
  --pgn-alert-icon-color: var(--pgn-color-alert-icon-info);
}

.alert-danger {
  --pgn-alert-bg: var(--pgn-color-alert-bg-danger);
  --pgn-alert-border-color: var(--pgn-color-alert-border-danger);
  --pgn-alert-icon-color: var(--pgn-color-alert-icon-danger);
}

.alert-warning {
  --pgn-alert-bg: var(--pgn-color-alert-bg-warning);
  --pgn-alert-border-color: var(--pgn-color-alert-border-warning);
  --pgn-alert-icon-color: var(--pgn-color-alert-icon-warning);
}
