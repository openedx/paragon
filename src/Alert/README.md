---
title: 'Alert'
type: 'component'
components:
- Alert
- AlertHeading
- AlertLink
categories:
- Status & metadata
status: 'Stable'
designStatus: 'Done'
devStatus: 'Done'
notes: |

---

The `Alert` component displays a short, important message in a way that attracts the user's attention.
Alerts offer four severity levels that set a distinctive icon and color:
- **Info**: used to convey general information or actions that aren't critical, note that info variant should not contain any icons in it
- **Warning**: used to display information that needs attention
- **Success**: used for success messages
- **Danger**: used to communicate problems that have to be resolved immediately

This component utilizes and extends the `Alert` component from react-bootstrap.<br/> <a href="https://react-bootstrap.github.io/components/alerts/" target="_blank" rel="noopener noreferrer"> See React-Bootstrap for additional documentation.</a>

## Basic Usage

```jsx live
<>
<Alert variant="success">
  This is a "success" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="info">
  This is a "info" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="danger">
  This is a "danger" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="warning">
  This is a "warning" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
</>
```

### With Action Buttons
The ``Alert`` component supports a dismissible button and a custom call-to-action button, via the ``dismissible`` and ``button`` props respectively. The buttons may be right aligned or stacked. On extra small screen widths, the buttons will be stacked.

The stacked variant should be used for:
- Placement in sidebars or small container
- Extra-small breakpoint and below
---

```jsx live
<>
  <Alert
    variant="info"
    actions={[
      <Button>Hello</Button>,
    ]}
    dismissible
    closeLabel="Dismiss"
    onClose={(e) => { console.log('closed', e); } }
  >
    This is a "info" alert with{' '}
    <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
    like.
  </Alert>
  <Alert variant="warning" dismissible>
    This is a "info" alert with{' '}
    <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
    like.
  </Alert>
  <Alert
    variant="success"
    icon={CheckCircle}
    dismissible
    actions={[
      <Button>Hello</Button>,
    ]}
  >
    <Alert.Heading>Hey, nice to see you</Alert.Heading>
    <p>
      Aww yeah, you successfully read this important alert message. This example
      text is going to run a bit longer so that you can see how spacing within an
      alert works with this kind of content.
    </p>
  </Alert>
  <Alert
    variant="danger"
    icon={Info}
    actions={[
      <Button>Hello</Button>,
    ]}
    dismissible
    onClose={(e) => { console.log('closed', e); } }
    stacked
  >
    <Alert.Heading>Hey, nice to see you</Alert.Heading>
    <p>
      Aww yeah, you successfully read this important alert message. This example
      text is going to run a bit longer so that you can see how spacing within an
      alert works with this kind of content.
    </p>
  </Alert>
</>
```

### Variants

```jsx live
<>
<Alert variant="success" dismissible icon={CheckCircle}>
  <Alert.Heading>Hey, nice to see you</Alert.Heading>
  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
</Alert>
<Alert variant="warning" dismissible icon={WarningFilled}>
  <Alert.Heading>Hey, nice to see you</Alert.Heading>
  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
</Alert>
<Alert variant="danger" dismissible icon={Info}>
  <Alert.Heading>Hey, nice to see you</Alert.Heading>
  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
</Alert>
</>
```

### Kitchen Sink

```jsx live
<Alert
  variant="success"
  actions={[
    <Button>Hello</Button>
  ]}
  dismissible
  stacked
>
  <Alert.Heading>Hey, nice to see you</Alert.Heading>
  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
  <hr />
  <p className="mb-0">
    Whenever you need to, be sure to use margin utilities to keep things nice
    and tidy.
  </p>
</Alert>
```
