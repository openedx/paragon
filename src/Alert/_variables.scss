// Alerts
//
// Define alert colors, border radius, and padding.

$alert-padding-y:                   1.5rem !default;
$alert-padding-x:                   1.5rem !default;
$alert-margin-bottom:               1rem !default;
$alert-border-radius:               $border-radius !default;
$alert-link-font-weight:            $font-weight-normal !default;
$alert-border-width:                0 !default;
$alert-title-color:                 #000000 !default;
$alert-box-shadow:                  0 1px 2px rgba(0, 0, 0, .15), 0 1px 4px rgba(0, 0, 0, .15) !default;

$alert-bg-level:                    -10 !default;
$alert-border-level:                -9 !default;
$alert-color-level:                 6 !default;

$alert-icon-space:                  .8rem !default;

$alert-font-size:                   .875rem !default;
$alert-line-height:                 1.5rem !default;
$alert-content-color:               $gray-700 !default;

$alert-actions-gap:                 map-get($spacers, 3);
