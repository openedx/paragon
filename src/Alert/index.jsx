import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import BaseAlert from 'react-bootstrap/Alert';
import divWithClassName from 'react-bootstrap/divWithClassName';
import { FormattedMessage } from 'react-intl';
import { useMediaQuery } from 'react-responsive';
import Icon from '../Icon';
import breakpoints from '../utils/breakpoints';
import Button from '../Button';
import ActionRow from '../ActionRow';

export const ALERT_CLOSE_LABEL_TEXT = 'Dismiss';

const Alert = React.forwardRef(({
  children,
  icon,
  actions,
  dismissible,
  onClose,
  closeLabel,
  stacked,
  ...props
}, ref) => {
  const [isStacked, setIsStacked] = useState(stacked);
  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth });
  const actionButtonSize = 'sm';

  useEffect(() => {
    if (isExtraSmall) {
      setIsStacked(true);
    } else {
      setIsStacked(stacked);
    }
  }, [isExtraSmall, stacked]);

  const cloneActionElement = useCallback(
    (Action) => {
      const addtlActionProps = { size: actionButtonSize, key: Action.props.children };
      return React.cloneElement(Action, addtlActionProps);
    },
    [],
  );

  return (
    <BaseAlert
      {...props}
      className={classNames('alert-content', props.className)}
      ref={ref}
    >
      {icon && <Icon src={icon} className="alert-icon" />}
      <div
        className={classNames({
          'pgn__alert-message-wrapper': !isStacked,
          'pgn__alert-message-wrapper-stacked': isStacked,
        })}
      >
        <div className="alert-message-content">
          {children}
        </div>
        {(dismissible || actions?.length > 0) && (
          <ActionRow className="pgn__alert-actions">
            <ActionRow.Spacer />
            {dismissible && (
              <Button
                size={actionButtonSize}
                variant="tertiary"
                onClick={onClose}
              >
                {closeLabel || (
                  <FormattedMessage
                    id="pgn.Alert.closeLabel"
                    defaultMessage="Dismiss"
                    description="Label of a close button on Alert component"
                  />
                )}
              </Button>
            )}
            {actions && actions.map(cloneActionElement)}
          </ActionRow>
        )}
      </div>
    </BaseAlert>
  );
});

// This is needed to display a default prop for Alert.Heading element
// Copied from react-bootstrap since BaseAlert.Heading component doesn't have defaultProps,
// so there seems to be no other way of providing correct default prop for base element
const DivStyledAsH4 = divWithClassName('h4');
DivStyledAsH4.displayName = 'DivStyledAsH4';

function AlertHeading(props) {
  return <BaseAlert.Heading {...props} />;
}
function AlertLink(props) {
  return <BaseAlert.Link {...props} />;
}

const commonPropTypes = {
  /** Specifies the base element */
  as: PropTypes.elementType,
  /** Overrides underlying component base CSS class name */
  bsPrefix: PropTypes.string,
};

AlertLink.propTypes = commonPropTypes;
AlertHeading.propTypes = commonPropTypes;

AlertLink.defaultProps = {
  as: 'a',
  bsPrefix: 'alert-link',
};

AlertHeading.defaultProps = {
  as: DivStyledAsH4,
  bsPrefix: 'alert-heading',
};

Alert.propTypes = {
  ...BaseAlert.propTypes,
  /** Docstring for the children prop */
  children: PropTypes.node,
  /** Docstring for the icon prop... Icon that will be shown in the alert */
  icon: PropTypes.func,
  /** Whether the alert is shown. */
  show: PropTypes.bool,
  /** Whether the alert is dismissible. Defaults to true. */
  dismissible: PropTypes.bool,
  /** Optional callback function for when the alert it dismissed. */
  onClose: PropTypes.func,
  /** Optional list of action elements. May include, at most, 2 actions, or 1 if dismissible is true. */
  actions: PropTypes.arrayOf(PropTypes.element),
  /** Position of the dismiss and call-to-action buttons. Defaults to ``false``. */
  stacked: PropTypes.bool,
  /** Sets the text for alert close button, defaults to 'Dismiss'. */
  closeLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};

Alert.defaultProps = {
  ...BaseAlert.defaultProps,
  children: undefined,
  icon: undefined,
  actions: undefined,
  dismissible: false,
  onClose: () => {},
  closeLabel: undefined,
  show: true,
  stacked: false,
};

Alert.Heading = AlertHeading;
Alert.Link = AlertLink;

export default Alert;
