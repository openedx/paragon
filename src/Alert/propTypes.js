import PropTypes from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const alertButtonProps = PropTypes.shape({
  buttonText: PropTypes.node.isRequired,
});
