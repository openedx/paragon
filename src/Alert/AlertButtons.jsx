import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from '../Button';
import { alertButtonProps } from './propTypes';

const AlertButtons = ({
  button,
  className,
  closeLabel,
  dismissible,
  onClose,
}) => {
  const buttonSize = 'sm';

  const [ctaButtonText, ctaButtonAttrs] = useMemo(
    () => {
      const buttonAttrs = {};
      const buttonText = button?.buttonText;
      if (button) {
        const {
          className: ctaButtonClassName,
          ...restButtonAttrs
        } = button;
        delete restButtonAttrs.buttonText;
        Object.assign(buttonAttrs, {
          size: buttonSize,
          className: classNames(ctaButtonClassName, { 'ml-3': dismissible }),
          ...restButtonAttrs,
        });
      }
      return [buttonText, buttonAttrs];
    },
    [JSON.stringify(button)],
  );

  return (
    <div className={classNames('pgn__alert-buttons d-flex justify-content-end', className)}>
      {dismissible && (
        <Button
          size={buttonSize}
          variant="tertiary"
          onClick={onClose}
        >
          {closeLabel}
        </Button>
      )}
      {button && (
        <Button {...ctaButtonAttrs}>
          {ctaButtonText}
        </Button>
      )}
    </div>
  );
};

AlertButtons.propTypes = {
  className: PropTypes.string,
  button: alertButtonProps,
  dismissible: PropTypes.bool,
  onClose: PropTypes.func,
  closeLabel: PropTypes.string,
};

AlertButtons.defaultProps = {
  className: undefined,
  button: undefined,
  dismissible: false,
  onClose: () => {},
  closeLabel: 'Dismiss',
};

export default AlertButtons;
