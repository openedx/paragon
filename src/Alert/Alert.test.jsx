import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer, { act } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Context as ResponsiveContext } from 'react-responsive';
import { Info } from '../../icons';
import breakpoints from '../utils/breakpoints';
import Button from '../Button';
import Alert from '.';

// eslint-disable-next-line react/prop-types
function AlertWrapper({ children, ...props }) {
  return (
    <IntlProvider locale="en" messages={{}}>
      <Alert {...props}>
        {children}
      </Alert>
    </IntlProvider>
  );
}

describe('<Alert />', () => {
  it('renders without any props', () => {
    const tree = renderer.create((
      <AlertWrapper>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with icon prop', () => {
    const tree = renderer.create((
      <AlertWrapper icon={Info}>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with dismissible prop', () => {
    const tree = renderer.create((
      <AlertWrapper dismissible>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('handles dismissible onClose', async () => {
    const mockOnClose = jest.fn();
    render(<AlertWrapper onClose={mockOnClose} dismissible>Alert</AlertWrapper>);
    const button = screen.getByRole('button');
    await userEvent.click(button);
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });
  it('renders with button prop', () => {
    const tree = renderer.create((
      <AlertWrapper actions={[<Button>Hello</Button>]}>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('handles button onClick', async () => {
    const mockOnClick = jest.fn();
    render(<AlertWrapper actions={[<Button onClick={mockOnClick}>Hello</Button>]}>Alert</AlertWrapper>);
    const button = screen.getByRole('button');
    await userEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
  it('renders with button and dismissible props', () => {
    const tree = renderer.create((
      <AlertWrapper actions={[<Button>Hello</Button>]} dismissible>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with stacked prop', () => {
    const tree = renderer.create((
      <AlertWrapper stacked actions={[<Button>Hello</Button>]} dismissible>Alert</AlertWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('switches to stacked variant at extra small breakpoint', () => {
    let tree;
    act(() => {
      tree = renderer.create((
        <ResponsiveContext.Provider value={{ width: breakpoints.extraSmall.maxWidth }}>
          <AlertWrapper dismissible>Alert</AlertWrapper>
        </ResponsiveContext.Provider>
      )).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });
});
