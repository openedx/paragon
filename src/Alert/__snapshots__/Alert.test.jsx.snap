// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders with button and dismissible props 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
    <div
      className="pgn__alert-actions pgn__action-row"
    >
      <span
        className="pgn__action-row-spacer"
      />
      <button
        className="btn btn-tertiary btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
      <button
        className="btn btn-primary btn-sm"
        disabled={false}
        type="button"
      >
        Hello
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders with button prop 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
    <div
      className="pgn__alert-actions pgn__action-row"
    >
      <span
        className="pgn__action-row-spacer"
      />
      <button
        className="btn btn-primary btn-sm"
        disabled={false}
        type="button"
      >
        Hello
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders with dismissible prop 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
    <div
      className="pgn__alert-actions pgn__action-row"
    >
      <span
        className="pgn__action-row-spacer"
      />
      <button
        className="btn btn-tertiary btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders with icon prop 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <span
    className="pgn__icon alert-icon"
  >
    <svg
      aria-hidden={true}
      fill="none"
      focusable={false}
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders with stacked prop 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <div
    className="pgn__alert-message-wrapper-stacked"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
    <div
      className="pgn__alert-actions pgn__action-row"
    >
      <span
        className="pgn__action-row-spacer"
      />
      <button
        className="btn btn-tertiary btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
      <button
        className="btn btn-primary btn-sm"
        disabled={false}
        type="button"
      >
        Hello
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders without any props 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      Alert
    </div>
  </div>
</div>
`;

exports[`<Alert /> switches to stacked variant at extra small breakpoint 1`] = `null`;
